tema14Ianuarie/src/com/sci/bv/tema14Ianuarie/Main.java

```diff
@@ -0,0 +1,133 @@
+                            "2. Sell product\n" +
+                            "3. Display daily sales report\n" +
+                            "4. Exit\n" +
+                            "\n" +
+                            "Choose an option:\n" +
+                            "-------------------------------------------------------------\n");
+            x = scanner.nextInt();
+            if (x == 1)  // If the user chooses first option
+            {
+                System.out.println(
+                        "-------------------------------------------------------------\n" +
+                                "Product type (1 - animal; 2 - vegetable):\n" +
+                                "-------------------------------------------------------------\n" +
+                                "\n");
+                x = scanner.nextInt();
+                while (x == 1) {
+                    Animlist = myList.addAnimalProduct(); //Add animal product and characteristics
+                    System.out.println(
+                            "-------------------------------------------------------------\n" +
+                                    "1.Add another animal product\n" +
+                                    "0.Go back to main menu\n" +
+                                    "-------------------------------------------------------------\n" +
+                                    "\n");
+                    x = scanner.nextInt();
+                }
+                while (x == 2) {
+                    Veglist = myList1.addVegetableProduct();//Add animal product and characteristics
+                    System.out.println("-------------------------------------------------------------\n" +
+                            "2.Add another vegetable product\n" +
+                            "0.Go back to main menu\n" +
+                            "-------------------------------------------------------------\n" +
+                            "\n");
+                    x = scanner.nextInt();
+                }
+
+            }
+            while (x == 2) {
+                System.out.println("-------------------------------------------------------------\n" +
+                        "ID   Quantity   Product's name      Optimal Temperature    Price    Validity    Weight\n");
+                myList.displayProduct(); //Display animal products and characteristics
+                System.out.println("-------------------------------------------------------------\n" +
+                        "ID   Quantity   Product's name      Vitamins List    Price    Validity    Weight\n");
+                myList1.displayProduct();//Display vegetable products and characteristics
+                System.out.println("Please write the ID product:");
+                int ID = scanner.nextInt();
+                System.out.println("Please write the quantity:");
+                int Qty = scanner.nextInt();
+                for (AnimalProduct a : Animlist) {
```

Comment:
if you created the StorageAnimalList class that handles the adding of animal product why not let it handle the selling as well.
You could create a method there to "public DailySales sellProductToConsole(int ID, int Qty)"
it would do exactly this for and return a DailySales if successful or null if not.
You could check here for null and if that's the case then try the same thing with the vegetable.